src/psycopmlutils/timeseriesflattener/data_integrity.py

```diff
@@ -0,0 +1,271 @@
+"""Code to generate data integrity and train/val/test drift reports."""
+from pathlib import Path
+from typing import Optional, Tuple
+
+import pandas as pd
+from deepchecks.tabular import Dataset, Suite
+from deepchecks.tabular.checks import (
+    CategoryMismatchTrainTest,
+    DatasetsSizeComparison,
+    FeatureLabelCorrelation,
+    FeatureLabelCorrelationChange,
+    IdentifierLabelCorrelation,
+    IndexTrainTestLeakage,
+    NewLabelTrainTest,
+    TrainTestLabelDrift,
+)
+from deepchecks.tabular.suites import data_integrity, train_test_validation
+from wasabi import Printer
+
+
+def check_feature_sets_dir(path: Path, nrows: Optional[int] = None) -> None:
+    """Runs Deepcheck data integrity and train/val/test drift checks for a
+    given directory containing train/val/test files. Expects the directory to
+    contain 3 csv files, where the files have ["train", "val", "test"]
+    somewhere in their name.
+
+    The resulting reports are saved to a sub directory as .html files.
+
+    Args:
+        path (Path): Path to a directory containing train/val/test files
+        nrows (Optional[int]): Whether to only load a subset of the data.
+        Should only be used for debugging.
+    """
+    msg = Printer(timestamp=True)
+    msg.info("Running data integrity checks...")
+
+    out_dir = path / "deepchecks"
+    if not out_dir.exists():
+        out_dir.mkdir()
+    # create subfolder for outcome specific checks
+    outcome_checks_dir = out_dir / "outcomes"
+    if not outcome_checks_dir.exists():
+        outcome_checks_dir.mkdir()
+
+    ###################
+    #### DATA INTEGRITY
+    ###################
+
+    # Only running data integrity checks on the training set to reduce the
+    # chance of any form of peaking at the test set
+    train_predictors, train_outcomes = load_split_predictors_and_outcomes(
+        path=path,
+        split="train",
+        include_id=False,
```

Comment:
Are we sure we don't want to include IDs here? Seems relevant for making duplicate more transparent checking.

Reply:
Are we sure we don't want to include IDs here? Seems relevant for duplicate checking.